hazelcast/src/main/java/com/hazelcast/sql/impl/client/SqlClientResult.java

```diff
@@ -379,4 +386,10 @@ private JetSqlRow getCurrentRow() {
             return new JetSqlRow(service.getSerializationService(), values);
         }
     }
+
+    public Boolean isInfiniteRows() {
+        synchronized (mux) {
```

Comment:
Shouldn't we await for the state to be set, like the other methods (iterator, getRowMetadata etc.)? If we are just returning a boolean field, why not make it just volatile?

BTW, I feel like there is some unnecessary ceremony around this class. We don't return the SQL result to the user before the execute response comes. So, I feel like we can get rid of most of the complexities of this await state thing.

I believe it was done this way because we were returning the SQL result to the user before the execute response comes in earlier versions of the SQL implementation, but it is no longer the case

Reply:
Switched to volatile. I've created an issue HZ-1339 to rethink the synchronization here. Maybe it is not needed at all now.